pyriemann/transferlearning.py

```diff
@@ -0,0 +1,175 @@
+                'Value transfer_coef must be included in [0, 1] (Got %d)'
+                % self.transfer_coef)
+
+        if isinstance(self.metric, str):
+            self.metric_mean = self.metric
+            self.metric_dist = self.metric
+        elif isinstance(self.metric, dict):
+            # check keys
+            for key in ['mean', 'distance']:
+                if key not in self.metric.keys():
+                    raise KeyError('metric must contain "mean" and "distance"')
+
+            self.metric_mean = self.metric['mean']
+            self.metric_dist = self.metric['distance']
+        else:
+            raise TypeError('metric must be dict or str')
+
+        if self.transfer_coef != 0:
+            if set(y) != set(y_source):
+                raise ValueError(
+                    f"classes in source domain must match classes in target \
+                    domain. Classes in source are {np.unique(y_source)} while \
+                    classes in target are {np.unique(y)}")
+
+        if sample_weight is not None:
+            if (sample_weight.shape != (X_source.shape[0], 1)) and \
+                                (sample_weight.shape != (X_source.shape[0],)):
+                raise ValueError("Parameter sample_weight should either be \
+                    None or an ndarray shape (n_matrices, 1)")
+
+        if X.shape[1:] != X_source.shape[1:]:
+            raise ValueError("X and X_source must have the same number of \
+                n_channels")
+
+        if X.shape[0] != y.shape[0]:
+            raise ValueError("X and y must be for the same number of \
+                matrices i.e. n_matrices")
+
+        if sample_weight is None:
+            sample_weight = np.ones(X_source.shape[0])
+
+        if not (X_source.shape[0] == y_source.shape[0]
+                == sample_weight.shape[0]):
+            raise ValueError("X and y must be for the same number of \
+                matrices i.e. n_matrices")
+
+        self.classes_ = np.unique(y_source)
+
+        if self.n_jobs == 1:
+            self.target_means_ = [
+                mean_covariance(X[y == ll], metric=self.metric_mean)
+                for ll in self.classes_]
+            self.source_means_ = [
+                mean_covariance(
+                    X_source[y_source == ll],
+                    metric=self.metric_mean,
+                    sample_weight=sample_weight[y_source == ll]
+                )
+                for ll in self.classes_]
+        else:
+            self.target_means_ = Parallel(n_jobs=self.n_jobs)(
+                delayed(mean_covariance)(X[y == ll], metric=self.metric_mean)
+                for ll in self.classes_)
+            self.source_means_ = Parallel(n_jobs=self.n_jobs)(
+                delayed(mean_covariance)(
+                    X_source[y_source == ll],
+                    metric=self.metric_mean,
+                    sample_weight=sample_weight[y_source == ll])
+                for ll in self.classes_)
```

Comment:
this branching is useless. When n_jobs =1 then joblib Parallel is a pure list comprehension

Reply:
Ok, same situation happend in Kmeans and MDM, I'll correct this one here and I'll remove the branching from these models in another PR.